tests/unit/expire.tcl

```diff
@@ -601,4 +601,125 @@ start_server {tags {"expire"}} {
            {del foo}
         }
     } {} {needs:repl}
+
+    test {EXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100
+        assert_equal [r EXPIRE foo 200 NX] 0
+        assert_range [r TTL foo] 50 100
+    } {}
+
+    test {EXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200 NX] 1
+        assert_range [r TTL foo] 100 200
+    } {}
+
+    test {EXPIRE with XX option on a key with ttl} {
+        r SET foo bar EX 100
+        assert_equal [r EXPIRE foo 200 XX] 1
+        assert_range [r TTL foo] 100 200
+    } {}
+
+    test {EXPIRE with XX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200 XX] 0
+        assert_equal [r TTL foo] -1
+    } {}
+
+    test {EXPIRE with GT option on a key with lower ttl} {
+        r SET foo bar EX 100
+        assert_equal [r EXPIRE foo 200 GT] 1
+        assert_range [r TTL foo] 100 200
+    } {}
+
+    test {EXPIRE with GT option on a key with higher ttl} {
+        r SET foo bar EX 200
+        assert_equal [r EXPIRE foo 100 GT] 0
+        assert_range [r TTL foo] 100 200
+    } {}
+
+    test {EXPIRE with GT option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200 GT] 0
+        assert_equal [r TTL foo] -1
+    } {}
+
+    test {EXPIRE with LT option on a key with higher ttl} {
+        r SET foo bar EX 100
+        assert_equal [r EXPIRE foo 200 LT] 0
+        assert_range [r TTL foo] 50 100
+    } {}
+
+    test {EXPIRE with LT option on a key with lower ttl} {
+        r SET foo bar EX 200
+        assert_equal [r EXPIRE foo 100 LT] 1
+        assert_range [r TTL foo] 50 100
+    } {}
+
+    test {EXPIRE with LT option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 100 LT] 1
+        assert_range [r TTL foo] 50 100
+    } {}
+
+    test {EXPIRE with LT and XX option on a key with ttl} {
+        r SET foo bar EX 200
+        assert_equal [r EXPIRE foo 100 LT XX] 1
+        assert_range [r TTL foo] 50 100
+    } {}
+
+    test {EXPIRE with LT and XX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200 LT XX] 0
+        assert_equal [r TTL foo] -1
+    } {}
+
+    test {EXPIRE with conflict options} {
+        catch {r EXPIRE foo 200 LT GT} e
+        set e
+    } {ERR GT and LT options at the same time are not compatible}
+
+    test {EXPIRE with conflict options} {
+        catch {r EXPIRE foo 200 NX GT} e
+        set e
+    } {ERR NX and XX, GT or LT options at the same time are not compatible}
+
+    test {EXPIRE with conflict options} {
```

Comment:
```suggestion
    test {EXPIRE with conflicting options} {
```

Reply:
Same comment repeats a few times below.

Reply:
Fixed by add conflicting commands name to comments